zebra-state/src/service/arbitrary.rs

```diff
@@ -209,7 +213,24 @@ pub async fn populated_state(
     }
 
     while let Some(rsp) = responses.next().await {
-        rsp.expect("blocks should commit just fine");
+        // Wait for the block result and the chain tip update,
+        // which both happen in a separate thread from this one.
+        rsp.expect("unexpected block commit failure");
+
+        // Allow the chain tip update
+        if let Err(timeout_error) = timeout(
+            CHAIN_TIP_UPDATE_WAIT_LIMIT,
+            chain_tip_change.wait_for_tip_change(),
+        )
```

Comment:
I added this wait for tip change and timeout to fix test concurrency bugs in b890114eb4f8470a4c6b0d741d718167f894d531 and 770f73a76e28f826d7d8d34499e193cffb0eec9b.